### PR TITLE
Bluetooth: controller: split: Port advanced scheduling 

### PR DESCRIPTION
**Pull request by cvinayak**
Port advanced scheduling implementation from legacy
controller.

This implementation schedules
- non-overlapping scan window when there are simultaneous
central connections active
- central connection establishment with similar connection
intervals be placed in a non-overlapping  group, temporally
- connection parameter request with calculated window offset
hints to have non-overlapping BLE radio events.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

tempsnip

### CODE REVIEW

**Comment by cvinayak** [comment]
@carlescufi @aescolar @alexeRadu @George-Stefan can I get some reviewers. This PR is open getting close to being a month old soon.

Added @joerchan and @kruithofa for FYI.

**Comment by mtpr-ot** [comment]
This is a quite complicated change - will need to assign time for reviewing, understanding and testing. I will schedule this for beginning of next week.

**Comment by George-Stefan** [comment]
> This is a quite complicated change - will need to assign time for reviewing, understanding and testing. I will schedule this for beginning of next week.

Same here

**Comment by carlescufi** [comment]
@cvinayak given that code freeze for 2.2 is already passed, I have no option but to move this to 2.3. @mtpr-ot and @George-Stefan please review this as soon as possible so we can merge it as soon as the merge window for 2.3 opens.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c
Comment (mtpr-ot) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c]: The prefix "s_" is nolonger used in the controller, as far as I know

File: subsys/bluetooth/controller/ll_sw/ull_conn.c
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_conn.c]: Please replace locations where conn_upd_curr is reset with a call to this function

File: subsys/bluetooth/controller/ll_sw/ull_conn_internal.h
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_conn_internal.h]: Will this function be called from other modules? If not it should be static.

**Review by kruithofa — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_sched.c
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_sched.c]: nitpick: elsewhere lower-case f is used
if 0xff is a flag, shouldn't we use a #define for this?

File: subsys/bluetooth/controller/ll_sw/ull_sched.c
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_sched.c]: use a #define instead of 1250; this number is also used other places in the code

File: subsys/bluetooth/controller/ll_sw/ull_sched.c
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_sched.c]: easily confused with the win_offset parameter. Maybe use another name, instead of just differing by underscore?

**Review by kruithofa — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_conn_internal.h
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_conn_internal.h]: This is called from ull_sched.c

File: subsys/bluetooth/controller/ll_sw/ull_sched.c
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_sched.c]: https://github.com/zephyrproject-rtos/zephyr/issues/23314

**Comment by cvinayak** [comment]
depends on https://github.com/zephyrproject-rtos/zephyr/pull/23391

**Comment by cvinayak** [comment]
Removed DNM, as no regression has been detected during all my development and testing that included the changes in this PR. (Any additional changes/Fixes are in #22237)

**Comment by cvinayak** [comment]
@carlescufi @mtpr-ot @kruithofa This PR has been around for 4 months now, can we get some approvals.

**Comment by carlescufi** [comment]
@kruithofa could you please re-review?
@cvinayak please rebase

**Review by kruithofa — APPROVED** [review]

**Review by mtpr-ot — APPROVED** [review]
LGTM

**Review by carlescufi — APPROVED** [review]

**Pull request merged**
